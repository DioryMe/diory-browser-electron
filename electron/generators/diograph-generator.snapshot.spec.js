const { join } = require('path')
const glob = require('glob')
const { v4: uuid } = require('uuid')
const { statSync } = require('fs')

const { generateDiograph } = require('./diograph-generator')
const { getDefaultImage } = require('../../src/shared/getDefaultImage')

jest.mock('uuid')
jest.mock('fs', () => ({
  ...jest.requireActual('fs'),
  statSync: jest.fn(),
}))
jest.mock('../../src/shared/getDefaultImage')

function getNumberOfFilesAndFolders(path) {
  return glob.sync(join(path, '/**/*')).length
}

function getArray(length) {
  return [...Array(length).keys()]
}

describe('diograph-generator', () => {
  describe('generateDiograph', () => {
    beforeEach(() => {
      const fileStats = {
        mtime: {
          toISOString: jest.fn().mockReturnValue('some-modified'),
        },
        birthtime: {
          toISOString: jest.fn().mockReturnValue('some-created'),
        },
      }
      statSync.mockReturnValue(fileStats)
      getDefaultImage.mockReturnValue('some-default-image')
    })
    it('generates diograph from example folder', async () => {
      const exampleFolderPath = join(__dirname, '../readers/example-folder')
      const amountOfCases = getNumberOfFilesAndFolders(exampleFolderPath) + 1
      getArray(amountOfCases).forEach((id) => {
        uuid.mockReturnValueOnce(`uuid-${id}`)
      })

      const { diograph } = await generateDiograph(exampleFolderPath)

      // Relative paths for image & contentUrl
      // TODO: Move to own function in generate-diograph-channel
      //       && call it from there instead of copy-pasting here
      Object.keys(diograph).forEach((dioryId) => {
        const diory = diograph[dioryId]
        if (diory.image) {
          diory.image = diory.image.replace(`${exampleFolderPath}/`, '')
        }
        if (diory.data && diory.data.contentUrl) {
          diory.data.contentUrl = diory.data.contentUrl.replace(`${exampleFolderPath}/`, '')
        }
      })

      expect(diograph).toMatchSnapshot()
    })
  })
})
