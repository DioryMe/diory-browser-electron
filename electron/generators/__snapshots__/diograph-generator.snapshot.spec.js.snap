// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diograph-generator generateDiograph generates diograph from example folder 1`] = `
Object {
  "uuid-0": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "VideoObject",
      "contentUrl": "some-video.mp4",
      "encodingFormat": "video/x-m4v",
    },
    "id": "uuid-0",
    "modified": "some-modified",
    "text": "some-video.mp4",
  },
  "uuid-1": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "DigitalDocument",
      "contentUrl": "some-document.pdf",
      "encodingFormat": "application/pdf",
    },
    "id": "uuid-1",
    "modified": "some-modified",
    "text": "some-document.pdf",
  },
  "uuid-10": Object {
    "created": "some-created",
    "date": "2012-05-25T06:10:32.666Z",
    "id": "uuid-10",
    "image": "image-with-24-hour.jpg",
    "latitude": 51.735696851851856,
    "links": Object {
      "image-with-24-hour.jpg": Object {
        "id": "uuid-4",
      },
      "some-document.docx": Object {
        "id": "uuid-7",
      },
      "some-document.odt": Object {
        "id": "uuid-3",
      },
      "some-document.pdf": Object {
        "id": "uuid-1",
      },
      "some-folder": Object {
        "id": "uuid-9",
      },
      "some-image.jpg": Object {
        "id": "uuid-5",
      },
      "some-music.mp3": Object {
        "id": "uuid-6",
      },
      "some-text.txt": Object {
        "id": "uuid-2",
      },
      "some-video.mp4": Object {
        "id": "uuid-0",
      },
    },
    "longitude": 12.688426296296297,
    "modified": "some-modified",
    "text": "example-folder",
  },
  "uuid-2": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "DigitalDocument",
      "contentUrl": "some-text.txt",
    },
    "id": "uuid-2",
    "modified": "some-modified",
    "text": "some-text.txt",
  },
  "uuid-3": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "DigitalDocument",
      "contentUrl": "some-document.odt",
      "encodingFormat": "application/vnd.oasis.opendocument.text",
    },
    "id": "uuid-3",
    "modified": "some-modified",
    "text": "some-document.odt",
  },
  "uuid-4": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ImageObject",
      "contentUrl": "image-with-24-hour.jpg",
      "encodingFormat": "image/jpeg",
      "height": 3000,
      "width": 4000,
    },
    "date": "2019-07-09T00:01:16.000Z",
    "id": "uuid-4",
    "image": "image-with-24-hour.jpg",
    "latitude": 68.27818055555555,
    "longitude": 14.302322222222223,
    "modified": "some-modified",
  },
  "uuid-5": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ImageObject",
      "contentUrl": "some-image.jpg",
      "encodingFormat": "image/jpeg",
      "height": 480,
      "width": 640,
    },
    "date": "2008-11-01T21:15:11.000Z",
    "id": "uuid-5",
    "image": "some-image.jpg",
    "latitude": 43.464455,
    "longitude": 11.881478333333334,
    "modified": "some-modified",
  },
  "uuid-6": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "AudioObject",
      "contentUrl": "some-music.mp3",
      "encodingFormat": "audio/mpeg",
    },
    "id": "uuid-6",
    "modified": "some-modified",
    "text": "some-music.mp3",
  },
  "uuid-7": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "DigitalDocument",
      "contentUrl": "some-document.docx",
      "encodingFormat": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    },
    "id": "uuid-7",
    "modified": "some-modified",
    "text": "some-document.docx",
  },
  "uuid-8": Object {
    "created": "some-created",
    "data": Object {
      "@context": "https://schema.org",
      "@type": "ImageObject",
      "contentUrl": "some-folder/some-folder-image.jpg",
      "encodingFormat": "image/jpeg",
      "height": 480,
      "width": 640,
    },
    "date": "2008-11-01T21:15:11.000Z",
    "id": "uuid-8",
    "image": "some-folder/some-folder-image.jpg",
    "latitude": 43.464455,
    "longitude": 11.881478333333334,
    "modified": "some-modified",
  },
  "uuid-9": Object {
    "created": "some-created",
    "date": "2008-11-01T21:15:11.000Z",
    "id": "uuid-9",
    "image": "some-folder/some-folder-image.jpg",
    "latitude": 43.464455,
    "links": Object {
      "some-folder-image.jpg": Object {
        "id": "uuid-8",
      },
    },
    "longitude": 11.881478333333334,
    "modified": "some-modified",
    "text": "some-folder",
  },
}
`;
